### PR TITLE
Bump PyVista for new DynamicScraper features

### LINKED ISSUE #606: Dynamically load interactive examples in documentation
## 🐛 Bug Report
Some of the documentation examples in the gallery are taking a bit of time to load for me and might make the examples unresponsive for people with low bandwidth.

## How to Reproduce
Load an example page like https://geovista.readthedocs.io/en/latest/generated/gallery/spatial_index/uber_h3.html and notice the ~45MB scene being immediately downloaded.

## Expected Behaviour
The scene could load dynamically upon request. PyVista's image scraper implements this for you! I thought it was the default, but maybe it was turned off here or a dependency is missing? Or perhaps this is new to 0.43.1 and the version of PyVista just needs to be bumped to get these tabs?

@bjlittle, was the choice to make all examples dynamic by default intentional?

What I'm thinking these scenes should look like:

https://github.com/bjlittle/geovista/assets/22067021/07b1cbd3-f883-4563-ae40-edfeb89b7838

### PR DESCRIPTION
**Pull request by banesullivan**
Resolves #606.

This bumps the version of PyVista and makes some minor changes so that we can leverage PyVista's latest features in its DynamicScraper

### CODE REVIEW

**Comment by banesullivan** [comment]
FYI @bjlittle, I noticed `render_points_as_spheres=True` is used some places here. This may display oddly in the docs as pointed out in https://github.com/MatthewFlamm/pyransame/pull/136#issuecomment-1883499716

**Comment by bjlittle** [comment]
@banesullivan Lovely! Thanks!

I just need to release `0.4.1` and merge back the changes in the `0.4.x` release feature branch to `main`.

Then you can rebase and we can bank this 🍻

**Review by bjlittle — CHANGES_REQUESTED** [review]
@banesullivan 

[0.4.1](https://github.com/bjlittle/geovista/releases/tag/v0.4.1) has now been released and the `0.4.x` feature branch has been merged back to `main` (#633).

Fancy rebasing from `main`, then we can bank this :+1:

Thanks!

File: docs/Makefile [hunk 1]
```diff
@@ -29,3 +29,9 @@ clean:
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
 	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+# Spin up a local http server to view the rendered documentation.
+# This is required for interactive examples to work.
+serve-html:
+	@echo "Serving documentation at http://localhost:11000"
+	@python -m http.server 11000 --directory "$(BUILDDIR)"/html
```
Comment (bjlittle) [inline docs/Makefile:37]: Nice! :star_struck:

**Comment by bjlittle** [comment]
> FYI @bjlittle, I noticed `render_points_as_spheres=True` is used some places here. This may display oddly in the docs as pointed out in [MatthewFlamm/pyransame#136 (comment)](https://github.com/MatthewFlamm/pyransame/pull/136#issuecomment-1883499716)

Is this using `vkt.js` under the hood to render? Just curious ...

I noticed that "text" also isn't supported e.g., titles or points labels.

**Comment by bjlittle** [comment]
@banesullivan @tkoyama010

O--M--G! 😍🚀💯  

I soooo love it! Amazing!
image

**Comment by bjlittle** [comment]
@banesullivan BTW is there a way to control the size of the `Interactive Scene` rendered window?

**Comment by tkoyama010** [comment]
@bjlittle 
Awesome! By the way, the preview of the RTD that you set up at #391 is not showing up in CI, do you know why?

**Comment by bjlittle** [comment]
> @bjlittle Awesome! By the way, the preview of the RTD that you set up at #391 is not showing up in CI, do you know why?

Yes. I had to temporarily disable it whilst I was working on the `v0.4.x` release feature branch. This is because the `v0.4.x` is based on `geovista` 0.4.0, and this was before the documentation was building successfully (after my recent work on it).

I didn't want failing RTD CI tasks on the `v0.4.x` pull-requests.

I've since enabled it again after deploying `0.4.1`.

**Comment by bjlittle** [comment]
Yup, that did the trick 👍

**Comment by tkoyama010** [comment]
@bjlittle Thanks. WFM.

**Comment by bjlittle** [comment]
The `pre-commit.ci` job will remain `Pending` until the merge conflict is resolved.

Neat. Didn't realise that `pre-commit.ci` detected this on a pull-request.

image

**Comment by banesullivan** [comment]
@bjlittle, the conflict is:

```
<<<<<<< banesullivan/bump-pyvista
  - pyvista >=0.40
=======
  - pyvista >=0.43.1
>>>>>>> main
```

Which do you want to support as the minimum version of PyVista?

**Comment by bjlittle** [comment]
Go with `>=0.43.1` please.

Thanks 👍

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Review by bjlittle — APPROVED** [review]
Awesome, thanks 🚀

**Comment by bjlittle** [comment]
@all-contributors please add @banesullivan for infra and examples

**Pull request merged**